src/main/java/wolox/training/repositories/UserRepository.java

```diff
@@ -2,10 +2,18 @@
 
 import java.time.LocalDate;
 import java.util.List;
+import java.util.Optional;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
 import wolox.training.models.User;
 
 public interface UserRepository extends JpaRepository<User, Long> {
   User findByUsername(String username);
-  List<User> findByNameContainingIgnoreCaseAndBirthdayBetween(String name, LocalDate startDate, LocalDate endDate);
+
+  @Query("select u from User u where u.name like %:name% and (u.birthday >= :startDate and u.birthday <= :endDate) or u.birthday is null")
```

Comment:
You're missing parentheses outside the whole birthdate thing:
((u.birthday >= :startDate and u.birthday <= :endDate) or u.birthday is null")